source/model/PadelMatch.mc

```diff
@@ -65,10 +68,24 @@ class PadelMatch {
         if (self.goldenPoint) {
             // mid game
             if (self.matchStatus.getP1Score() != 40) {
+                if (self.matchStatus.getP1Score() == 'A') {
```

Comment:
this sounds weird. because we check for golden point, and now we need to check for advantages. seems confusing

if we moved into the 3 options (golden vs adv vs amateur) we could keep the path for each 'style' of game here, which would align nicely with the idea that I have to refactor this code to have a sort of game engine class, with subclasses that would know how to compute the score (actually 1 match engine, containing set engines, containing game engine. something like that)

Reply:
Same as before, if you plan to fix #27 so any "game mode" can be independent, in this case I used your basic logic and I tried to not change it too much, so I think this's the only way I found, that is easy with one extra check, that is needed I think.
What do you think?